x-pack/plugins/security_solution/public/common/components/navigation/tab_navigation/index.tsx

```diff
@@ -8,60 +8,36 @@
 import { EuiTab, EuiTabs } from '@elastic/eui';
 import { getOr } from 'lodash/fp';
 import React, { useEffect, useState, useCallback, useMemo } from 'react';
-import { useHistory } from 'react-router-dom';
+import { useHistory, useLocation } from 'react-router-dom';
 import deepEqual from 'fast-deep-equal';
 
-import { APP_ID } from '../../../../../common/constants';
 import { track, METRIC_TYPE, TELEMETRY_EVENT } from '../../../lib/telemetry';
-import { getSearch } from '../helpers';
 import { TabNavigationProps, TabNavigationItemProps } from './types';
-import { useKibana } from '../../../lib/kibana';
-import { SecurityPageName } from '../../../../app/types';
-import { useFormatUrl } from '../../link_to';
 
 const TabNavigationItemComponent = ({
   disabled,
-  href,
   hrefWithSearch,
   id,
   name,
   isSelected,
-  pageId,
-  urlSearch,
 }: TabNavigationItemProps) => {
   const history = useHistory();
-  const { navigateToApp, getUrlForApp } = useKibana().services.application;
-  const { formatUrl } = useFormatUrl(((pageId ?? id) as unknown) as SecurityPageName);
   const handleClick = useCallback(
     (ev) => {
       ev.preventDefault();
-      if (id in SecurityPageName && pageId == null) {
```

Comment:
Internal tab navigation doesn't require all this logic because it only navigates between tabs of a Sub-plugin and not between Sub-plugins.